pilot/pkg/bootstrap/server.go

```diff
@@ -535,6 +699,36 @@ func (s *Server) initEventHandlers() error {
 	return nil
 }
 
+// add a GRPC listener using DNS-based certificates. Will be used for Galley, injection and CA signing.
+func (s *Server) initDNSListener(args *PilotArgs) error {
+	// Create k8s-signed certificates. This allows injector, validation to work without Citadel, and
+	// allows secure SDS connections to Istiod.
+	err := s.initDNSCerts()
+	if err != nil {
+		log.Warna("Failed to generate DNS k8s-signed certs. Autoinjection and validation will not be enabled, and"+
+			"port 15012 will not be activated", err)
```

Comment:
```suggestion
		log.Warnf("Failed to generate DNS k8s-signed certs. Autoinjection and validation will not be enabled, and "+
			"port 15012 will not be activated: %v", err)
```

missing spaces right now.

Specifically I noticed this from failure `Failed to create certs dir: mkdir ./var/run/secrets/istio-dns: permission denied` is this expected